app/models/competition.rb

```diff
@@ -2548,6 +2548,8 @@ def set_form_data(form_data, current_user)
       end
 
       assign_attributes(Competition.form_data_to_attributes(form_data))
+
+      self.uses_v2_registrations = false unless form_data.dig('website', 'usesWcaRegistration')
```

Comment:
At this point, the attributes from the form data have already been assigned (in the line directly above), so you might as well do `unless self.use_wca_registration`.

Reply:
Right you are!